flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutor.java

```diff
@@ -1012,10 +1016,34 @@ public void disconnectResourceManager(Exception cause) {
 		}
 	}
 
+	@Override
+	public CompletableFuture<ThreadDumpInfo> requestThreadDump(Time timeout) {
+		final Collection<ThreadInfo> threadDump = JvmUtils.createThreadDump();
+
+		final Collection<ThreadDumpInfo.ThreadInfo> threadInfos = threadDump.stream()
+			.map(threadInfo -> ThreadDumpInfo.ThreadInfo.create(threadInfo.getThreadName(), threadInfo.toString()))
```

Comment:
I'm wondering whether this conversion should really be done on the TaskExecutor.
I'd prefer if conversions to REST responses are done by the handler just to separate things a bit.

Reply:
As soon as the RestResponse and the internal cluster message differ I will introduce a separate the classes. As long as they are identical I think it is fine to use the same class here.